frontend/src/utils/tests/geoFileFunctions.test.js

```diff
@@ -0,0 +1,135 @@
+import fs from 'fs';
+import {
+  twoFeaturesOneInvalid,
+  validMultiPolygonAoi,
+  emptyMultiPolygon,
+  polygonAndMultiPolygon,
+} from './snippets/AOI';
+import {
+  PolygonTypeFeature,
+  LineStringTypeFeature,
+  MultiPolygonTypeFeature,
+  closedLinestringFeature,
+  convertedLineStringToPolygonFeature,
+} from './snippets/feature';
+
+import { parsedGeometry } from './snippets/parsedGeometry';
+
+import {
+  verifyGeometry,
+  verifyFileSize,
+  validateFeature,
+  readGeoFile,
+  verifyFileFormat,
+} from '../geoFileFunctions';
+
+describe('verifyGeometry', () => {
+  const supportedGeoms = ['Polygon', 'MultiPolygon'];
+  it('with an empty MultiPolygon, should return no features', () => {
+    expect(verifyGeometry(emptyMultiPolygon, supportedGeoms).features.length).toBe(0);
+  });
+  it('with a valid MultiPolygon geometry', () => {
+    expect(verifyGeometry(validMultiPolygonAoi, supportedGeoms)).toEqual(validMultiPolygonAoi);
+  });
+  it('with a polygon and a MultiPolygon', () => {
+    expect(verifyGeometry(polygonAndMultiPolygon, supportedGeoms)).toEqual(polygonAndMultiPolygon);
+  });
+  it('with an aoi containing an empty MultiPolygon, should exclude the invalid geometry', () => {
+    expect(twoFeaturesOneInvalid.features.length).toBe(2);
+    expect(verifyGeometry(twoFeaturesOneInvalid, supportedGeoms).features.length).toBe(1);
+  });
+  it('throws an error if geometry is not a FeatureCollection', () => {
+    expect(() => verifyGeometry(PolygonTypeFeature)).toThrow('noFeatureCollection');
+  });
+});
+
+describe('verifyFileSize', () => {
+  it('throws an error when the file is too large', () => {
+    let file = new File([new ArrayBuffer(1048576)], 'file.json');
```

Comment:
As it depends on the MAX_FILESIZE config value, it would be better to use that var and sum some bytes more to it.